src/main/java/com/github/bartimaeusnek/cropspp/crops/gregtechCrops/StonelillyCrop.java

```diff
@@ -55,95 +55,67 @@ public String name() {
+        // if crop is on last stage, it needs the block
+        switch (color) {
+            case "Red":
+                return crop.isBlockBelow("stoneGraniteRed") || crop.isBlockBelow("blockGranite");
+            case "Black":
+                return crop.isBlockBelow("stoneGraniteBlack") || crop.isBlockBelow("stoneBasalt");
+            case "White":
+                return crop.isBlockBelow("blockMarble") || crop.isBlockBelow("blockDiorite");
+            case "Gray":
+                return crop.isBlockBelow(Blocks.cobblestone) || crop.isBlockBelow(Blocks.stone)
+                        || crop.isBlockBelow("blockAndesite");
+            case "Yellow":
+                return crop.isBlockBelow(Blocks.end_stone) || crop.isBlockBelow(Blocks.sand)
+                        || crop.isBlockBelow(Blocks.sandstone);
+            case "Nether":
+                return crop.isBlockBelow(Blocks.netherrack) || crop.isBlockBelow(Blocks.nether_brick);
+            default:
+                // if this line executes consider it a UB
+                return false;
         }
-        return ret;
     }
 
     @Override
     public int weightInfluences(ICropTile crop, float humidity, float nutrients, float air) {
-        return (int) ((double) humidity * 0.8 + (double) nutrients * 1.4 + (double) air * 0.8);
+        return (int) ((double) humidity / 0.8D + (double) nutrients / 1.4D + (double) air / 0.8D);
     }
 
     @Override
     public String[] attributes() {
-        String[] ret = null;
         switch (color) {
-            case "Red": {
-                ret = new String[] { color, "Stone", "Fire" };
-                break;
-            }
-            case "Black": {
-                ret = new String[] { color, "Stone", "Dark" };
-                break;
-            }
-            case "White": {
-                ret = new String[] { color, "Stone", "Shiny" };
-                break;
-            }
-            case "Gray": {
-                ret = new String[] { color, "Stone", "Metal" };
-                break;
-            }
-            case "Yellow": {
-                ret = new String[] { color, "Stone", "Alien" };
-                break;
-            }
-            case "Nether": {
-                ret = new String[] { color, "Stone", "Evil" };
-                break;
-            }
+            case "Red":
+                return new String[] { color, "Stone", "Fire" };
+            case "Black":
+                return new String[] { color, "Stone", "Dark" };
+            case "White":
+                return new String[] { color, "Stone", "Shiny" };
+            case "Gray":
+                return new String[] { color, "Stone", "Metal" };
+            case "Yellow":
+                return new String[] { color, "Stone", "Alien" };
+            case "Nether":
+                return new String[] { color, "Stone", "Evil" };
+            default:
+                // if this line executes consider it a UB
+                return null;
```

Comment:
Probably return non-null for safety

Reply:
should have been addressed by https://github.com/GTNewHorizons/Crops-plus-plus/pull/57/commits/38c9838a209ddeed10cd00bfea8786ac8a2a3baf